### PR TITLE
Kopets ToDoList

### PR DESCRIPTION
**Pull request by sidgar21**
Creating ToDoList with fixed bugs

### CODE REVIEW

File: Liubomyr_Kopets/ToDo/src/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Component } from '@angular/core';
+
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/app.component.ts:2]: Remove extra line. Try to keep only one empty line between blocks of code.

File: Liubomyr_Kopets/ToDo/src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/app.component.ts:7]: Remove extra line. Try to keep only one empty line between blocks of code.

File: Liubomyr_Kopets/ToDo/src/app/app.module.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+
+import { AppRoutingModule } from './app-routing.module';
+import { AppComponent } from './app.component';
+import { TodocomponentComponent } from './todocomponent/todocomponent.component';
+import { TodolistsComponent } from './todolists/todolists.component';
+
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/app.module.ts]: Remove extra line. Try to keep only one empty line between blocks of code.

File: Liubomyr_Kopets/ToDo/src/app/todocomponent/todocomponent.component.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+@Component({
+  selector: 'app-todocomponent',
+  templateUrl: './todocomponent.component.html',
+  styleUrls: ['./todocomponent.component.scss']
+})
+export class TodocomponentComponent {
+  @Input() items: string;
+  @Output() editItem = new EventEmitter();
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/todocomponent/todocomponent.component.ts]: Add type for each EventEmitter(generic)

File: Liubomyr_Kopets/ToDo/src/app/todolists/todolists.component.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+})
+export class TodolistsComponent {
+
+  title = 'todo';
+  inputText: string;
+  text: string;
+  list: string[] = [];
+  isEdit = false;
+  btnAdd = 'Add';
+  i: number;
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/todolists/todolists.component.ts]: The name of the property isn't clear.

File: Liubomyr_Kopets/ToDo/src/app/todolists/todolists.component.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  styleUrls: ['./todolists.component.scss']
+})
+export class TodolistsComponent {
+
+  title = 'todo';
+  inputText: string;
+  text: string;
+  list: string[] = [];
+  isEdit = false;
+  btnAdd = 'Add';
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/todolists/todolists.component.ts]: Remove unused properties

File: Liubomyr_Kopets/ToDo/src/app/todolists/todolists.component.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-todolists',
+  templateUrl: './todolists.component.html',
+  styleUrls: ['./todolists.component.scss']
+})
+export class TodolistsComponent {
```
Comment (Magfiozy) [inline Liubomyr_Kopets/ToDo/src/app/todolists/todolists.component.ts]: Do you have a lot of lists? If no use singular, please.

File: Liubomyr_Kopets/ToDo/src/app/todocomponent/todocomponent.component.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'app-todocomponent',
+  templateUrl: './todocomponent.component.html',
+  styleUrls: ['./todocomponent.component.scss']
+})
+export class TodocomponentComponent {
+  @Input() items: string;
+  @Output() editItem = new EventEmitter<any>();
```
Comment (mefissto) [inline Liubomyr_Kopets/ToDo/src/app/todocomponent/todocomponent.component.ts]: Avoid explicit setting the "any" type. If the event doesn't emit any value you can set "void" type

**Review by imotality — DISMISSED** [review]
Everything looks great!

**Review by LTania — DISMISSED** [review]

**Review by imotality — APPROVED** [review]
everything is OK

**Review by imotality — DISMISSED** [review]
good!

**Review by taraswithjazz — DISMISSED** [review]

File: Liubomyr_Kopets/ToDo/src/app/itemcomp/itemcomp.component.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-itemcomp',
+  templateUrl: './itemcomp.component.html',
+  styleUrls: ['./itemcomp.component.scss']
+})
+export class ItemcompComponent {
```
Comment (NazarLuchyt) [inline Liubomyr_Kopets/ToDo/src/app/itemcomp/itemcomp.component.ts]: Change to the correct name

File: Liubomyr_Kopets/ToDo/src/app/todolist/todolist.component.html [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  <div>
+    <button class="btn btn-success" [disabled]= "!inputText" (click)="addToList()" type="button">{{isEdit ? 'Save' : 'Add'}}</button>
+  </div>
+  <div>
+    <input type="text" [(ngModel)]="inputText" placeholder="Введіть текст">
+  </div>
+  <div>
+    <ul>
+      <li *ngFor="let item of itemsList; let itemIndex=index">
+       <app-itemcomp [item]="item" (editItem)="itemEdit(itemIndex)" (delItem)="itemDel(itemIndex)"(checkItem)="itemCheck(itemIndex)"></app-itemcomp>
```
Comment (NazarLuchyt) [inline Liubomyr_Kopets/ToDo/src/app/todolist/todolist.component.html]: Are you so lazy to write the full word 'delete'? Please work harder

File: Liubomyr_Kopets/ToDo/src/app/todolist/todolist.component.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  styleUrls: ['./todolist.component.scss']
+})
+export class TodolistComponent {
+
+  title = 'todo';
+  inputText: string;
+  itemsList: string[] = [];
+  isEdit = false;
+  indexEdit: number;
+  addToList() {
```
Comment (NazarLuchyt) [inline Liubomyr_Kopets/ToDo/src/app/todolist/todolist.component.ts]: Your function should only add to the list, why does it change list too?

**Review by kblueberry — DISMISSED** [review]

**Review by kblueberry — APPROVED** [review]

**Review by LTania — APPROVED** [review]

**Review by taraswithjazz — APPROVED** [review]

**Review by imotality — APPROVED** [review]

**Review by KolosUK — APPROVED** [review]

**Review by mefissto — APPROVED** [review]

**Pull request merged**